### PR TITLE
Support DragonFlyBSD

### PR DESCRIPTION
**Pull request by mazocomp**
Note this support is not complete yet, DragonFlyBSD doesn't have <term.h>

### CODE REVIEW

**Comment by mazocomp** [comment]
My bad, it does have <term.h>, I should manually specify path to ncurses include dir.

**Comment by mazocomp** [comment]
@hboetes I think this change is worth tagging so I can painlessly add this editor to Ravenports.

**Comment by hboetes** [comment]
@mazocomp it appears you missed my previous comment.

**Comment by mazocomp** [comment]
@hboetes which one?

**Comment by hboetes** [comment]
The one to the Makefile

**Comment by mazocomp** [comment]
@hboetes I can't find it.

**Comment by hboetes** [comment]
image

**Comment by mazocomp** [comment]
@hboetes I didn't receive notification about that commend nor I do see it. "Pending" might mean your comment is not sent.

I was thinking about the same thing, but I don't know how to execute shell commands in BSD Makefile.

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ SRCS=	autoexec.c basic.c bell.c buffer.c cinfo.c dir.c display.c \
 #
 # More or less standalone extensions.
 #
-SRCS+=	cmode.c cscope.c dired.c grep.c tags.c
+SRCS+=	cmode.c cscope.c dired.c grep.c tags.c reallocarray.c
```
[the remark this answers is not in the document]
Comment (hboetes) [reply Makefile]: You can do that with !=
For example:

PID!=           echo $$$$                                                                                                                                                     
REALIMAGE!=     echo /var/tmp/image.${PID}

**Comment by mazocomp** [comment]
@hboetes thank you, I am done!

**Comment by mazocomp** [comment]
@hboetes what do you think about using pkg-config(8)? OpenBSD's pkg-config(8) can't find both curses and ncurses :(

**Comment by hboetes** [comment]
@mazocomp this pull request start to look real nice. I’ll give it a few more days for reconsideration.

Who would want to use this port on OpenBSD, mg is preinstalled? In the GNUMakefile there already is pkg-config included.

**Comment by mazocomp** [comment]
@hboetes in that case I should do something with DPADD variable, I still don't know what's exactly going on, if we leave it as is, that'll make mg depend on private ncurses in DragonFly's base: ${DESTDIR}${LIBDIR}/priv/libprivate_ncurses.a

**Comment by mazocomp** [comment]
@hboetes ok, I don't care about DPADD variable, let's hope this last diff satisfies every OS except OpenBSD.

**Comment by hboetes** [comment]
I hope you have a bit of patience. I’m on vacation for 2 weeks now.

**Comment by mazocomp** [comment]
@hboetes I don't have patience at all, but in this case I don't care, I have many other problems.

**Comment by hboetes** [comment]
Done! Thanks for your hard work!

**Comment by hboetes** [comment]
And I also added a tag. :-)

**Comment by mazocomp** [comment]
@hboetes what do you think about pruning GNUmakefile? Linux folks can use BSD make anyway.

**Comment by hboetes** [comment]
No way. BSD Make is an extra dependency, besides it doesn’t work. This
GNUMakefile works great.

**Comment by mazocomp** [comment]
@hboetes ok, Ravenports creates CMakeLists.txt in this port anyway :(

**Comment by hboetes** [comment]
A CMakefile would be nice though :-)

**Comment by mazocomp** [comment]
@hboetes ok, I'll send few patches with CMakeLists.txt today

**Pull request merged**